### PR TITLE
[GH-2525] bug: Make ST_Force3D return multipolygons even if it's a single polygon inside

### LINKED ISSUE #2525: bug: ST_Force3d() returns polygon when input is multipolygon with a single nested polygon
See https://github.com/apache/sedona/pull/2512#issuecomment-3573577084. Basically, we return polygon instead of multipolygon.

### PR DESCRIPTION
**Pull request by petern48**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest/community/develop/)

## Is this PR related to a ticket?

- Yes, and the PR name follows the format `[GH-XXX] my subject`. Closes #2525

## What changes were proposed in this PR?
When ST_Force3D received a MultiPolygon as input, it would previously downgrade it into a Polygon if possible. This PR makes it return a multipolygon even if it's a single polygon inside, which is the same behavior as Postgis.

## How was this patch tested?
Added tests

## Did this PR include necessary documentation updates?

- No, this is a minor edge case

### CODE REVIEW

File: common/src/main/java/org/apache/sedona/common/utils/GeometryForce3DTransformer.java [hunk 1]
```diff
@@ -49,4 +49,16 @@ public static Geometry transform(Geometry geometry, double zValue) {
     GeometryForce3DTransformer transformer = new GeometryForce3DTransformer(zValue);
     return transformer.transform(geometry);
   }
+
+  protected Geometry transformMultiPolygon(MultiPolygon geom, Geometry parent) {
```
Comment (petern48) [inline common/src/main/java/org/apache/sedona/common/utils/GeometryForce3DTransformer.java:53]: This is simply an override of this [method](https://github.com/locationtech/jts/blob/ab57bffd4250f14416315249565e585a76c0c489/modules/core/src/main/java/org/locationtech/jts/geom/util/GeometryTransformer.java#L283) in JTS's `GeometryTransformer` class to handle MultiPolygons separately.

**Comment by petern48** [comment]
R Build failure is unrelated. Looks like it's already in the process of being fixed: https://github.com/apache/sedona/pull/2524
